\newpage

## 1.

This is a binary classification problem, with the goal of predicting the presence of heart disease based on the patient's medical records (converting the original response variable num to binary: 0 represents no heart disease, 1 represents heart disease)

## 2.

The qualitative and quantitative variables in the data have been clearly distinguished and do not require conversion. Subsequently, according to the analysis needs, the target variable can be changed from a multi class variable to a binary class variable

## 3.

In [7]:
from ucimlrepo import fetch_ucirepo 
# fetch dataset 
heart_disease = fetch_ucirepo(id=45) 
import os
os.environ['OMP_NUM_THREADS'] = '1'
os.environ['LOKY_MAX_CPU_COUNT'] = '4'
import warnings
warnings.filterwarnings("ignore", category=UserWarning, 
                        module="sklearn.cluster._kmeans")
warnings.filterwarnings("ignore", category=UserWarning,
                        module="joblib.externals.loky.backend.context")

In [8]:
import pandas as pd
df = pd.concat([heart_disease.data.features, heart_disease.data.targets], axis=1)
print(df.head())

   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   1       145   233    1        2      150      0      2.3      3   
1   67    1   4       160   286    0        2      108      1      1.5      2   
2   67    1   4       120   229    0        2      129      1      2.6      2   
3   37    1   3       130   250    0        0      187      0      3.5      3   
4   41    0   2       130   204    0        2      172      0      1.4      1   

    ca  thal  num  
0  0.0   6.0    0  
1  3.0   3.0    2  
2  2.0   7.0    1  
3  0.0   3.0    0  
4  0.0   3.0    0  


In [9]:
print(df.dtypes)
print(df.shape[0])
print(df.shape[1])
print(df.describe())
print(df.head())

age           int64
sex           int64
cp            int64
trestbps      int64
chol          int64
fbs           int64
restecg       int64
thalach       int64
exang         int64
oldpeak     float64
slope         int64
ca          float64
thal        float64
num           int64
dtype: object
303
14
              age         sex          cp    trestbps        chol         fbs  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean    54.438944    0.679868    3.158416  131.689769  246.693069    0.148515   
std      9.038662    0.467299    0.960126   17.599748   51.776918    0.356198   
min     29.000000    0.000000    1.000000   94.000000  126.000000    0.000000   
25%     48.000000    0.000000    3.000000  120.000000  211.000000    0.000000   
50%     56.000000    1.000000    3.000000  130.000000  241.000000    0.000000   
75%     61.000000    1.000000    4.000000  140.000000  275.000000    0.000000   
max     77.000000    1.000000    4.000000  200.0000

The dataset contains 303 observations and 14 variables.There are some missing values in variables ca and tahl.The standard deviation of the chol variable is relatively large, and there may be extreme or outlier values far from the mean